app/code/Magento/CatalogSearch/view/frontend/templates/advanced/form.phtml

```diff
@@ -126,7 +126,8 @@
                        value="<?= $block->escapeHtml($block->getAttributeValue($_attribute)) ?>"
                        title="<?= $block->escapeHtml($block->getAttributeLabel($_attribute)) ?>"
                        class="input-text <?= $block->escapeHtmlAttr($block->getAttributeValidationClass($_attribute)) ?>"
-                       maxlength="<?= $block->escapeHtmlAttr($maxQueryLength) ?>" />
+                       maxlength="<?= $block->escapeHtmlAttr($maxQueryLength) ?>"
+                       data-validate="{'validate-alphanum':true}"/>
```

Comment:
Hi @RonakParmar. Such validation is not correct, you should check that entered string is not empty after it is trimmed instead.

What's the problem with currently observed result?

Reply:
@orlangur Currently if you enter only blank spaces in any of the fields then it will generate search result of it, which is not correct, So, I have done validation if user enters only spaces and click the Search button it will validate the input field and throw an error to enter valid string. I have fixed it in my local setup, please have a look.
![#23055-fixed-issue](https://user-images.githubusercontent.com/6551912/58628034-4607e900-82f6-11e9-8901-6c16cba008ee.png)